File: notewizback/NoteWiz/src/NoteWiz.Core/Entities/Note.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using NoteWiz.Core.Interfaces;

namespace NoteWiz.Core.Entities
{
    /// <summary>
    /// Stores notes created by users
    /// </summary>
    public class Note : IEntity
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(255)]
        public string Title { get; set; } = string.Empty;

        public string? Content { get; set; }

        [Required]
        public int UserId { get; set; }

        public int? DocumentId { get; set; }

        [Required]
        [MaxLength(14)]
        public string Color { get; set; } = "#FFFFFF";

        public bool IsPinned { get; set; }

        public bool IsPrivate { get; set; } = true; // Default to private

        [NotMapped]
        public string? CoverImage
        {
            get => CoverImageUrl;
            set => CoverImageUrl = value;
        }
        public string? CoverImageUrl { get; set; }

        public List<string> Tags { get; set; } = new();

        public int? CategoryId { get; set; }

        public string? Summary { get; set; }

        [ForeignKey("CategoryId")]
        public Category Category { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime? UpdatedAt { get; set; }

        public bool IsSynced { get; set; } // For mobile sync status
        public DateTime? LastSyncedAt { get; set; }

        public string? PageType { get; set; }

        public int? FolderId { get; set; }
        public virtual Folder Folder { get; set; }
        public virtual ICollection<FolderNote> FolderNotes { get; set; } = new List<FolderNote>(); // Many-to-many

        public bool IsOffline { get; set; }
        public DateTime? LastModifiedAt { get; set; }
        public string? DeviceId { get; set; }
        public string? SyncStatus { get; set; } // "synced", "pending", "conflict"

        // Navigation properties
        [ForeignKey("UserId")]
        public virtual User User { get; set; }

        [ForeignKey("DocumentId")]
        public virtual Document Document { get; set; }

        public virtual ICollection<NoteShare> SharedWith { get; set; }
        public virtual ICollection<NoteDrawing> NoteDrawings { get; set; }
        public virtual ICollection<NoteImage> NoteImages { get; set; }

        public Note()
        {
            Tags = new List<string>();
            SharedWith = new HashSet<NoteShare>();
            NoteDrawings = new HashSet<NoteDrawing>();
            NoteImages = new HashSet<NoteImage>();
            CreatedAt = DateTime.UtcNow;
            LastModifiedAt = DateTime.UtcNow;
            SyncStatus = "synced";
            IsOffline = false;
            IsPrivate = true; // Default to private
        }
    }
} 

File: notewizreact/nwprototype/src/types/note.ts
import { User, Task, Category, NavigationProps } from '../types';

export interface Note {
    id: string | number;
    title: string;
    content: string;
    coverId?: string;
    coverColor?: string;
    coverImage?: string;
    isPinned: boolean;
    isArchived: boolean;
    createdAt: string;
    updatedAt: string;
    userId: number;
    tags: string[];
    collaborators: User[];
    category?: string;
    isImportant?: boolean;
    folderId?: string | null;
    isPdf?: boolean;
    pdfUrl?: string;
    pdfName?: string;
}

export interface CreateNoteDto {
    title: string;
    content: string;
    coverId?: string;
    coverColor?: string;
    isArchived?: boolean;
<<<<<<< HEAD
    pageType?: string; // lined, grid, plain
=======
>>>>>>> 2919ceb5cf3c0d83b6677f30839892951700aa7c
}

export interface UpdateNoteDto {
    title: string;
    content: string;
    coverId?: string;
    coverColor?: string;
    isArchived?: boolean;
<<<<<<< HEAD
    pageType?: string; // lined, grid, plain
=======
>>>>>>> 2919ceb5cf3c0d83b6677f30839892951700aa7c
}

export interface NoteData {
    title: string;
    content: string;
    tags?: string[];
<<<<<<< HEAD
=======
    color?: string;
>>>>>>> 2919ceb5cf3c0d83b6677f30839892951700aa7c
    isImportant?: boolean;
    categoryId?: string;
    folderId?: string | null;
}

export interface NoteDto {
    id: number;
    title: string;
    content: string;
    coverId?: string;
    coverColor?: string;
    isArchived: boolean;
    createdAt: string;
    updatedAt?: string;
    userId: number;
}